scripts/rollup/bundles.js

```diff
@@ -120,7 +119,33 @@ const bundles = [
       'src/renderers/shared/**/*.js',
       'src/test/**/*.js', // ReactTestUtils is currently very coupled to DOM.
 
-      'src/isomorphic/classic/types/checkPropTypes.js',
+      'src/ReactVersion.js',
+      'src/shared/**/*.js',
+    ],
+  },
+  {
+    babelOpts: babelOptsReact,
+    bundleTypes: [NODE_DEV],
+    config: {
+      destDir: 'build/',
+      globals: {
+        react: 'React',
+      },
+      moduleName: 'ReactTestUtils',
+      sourceMap: false,
+    },
+    entry: 'src/test/ReactTestUtils.js',
+    externals: ['prop-types', 'prop-types/checkPropTypes', 'react'],
+    hasteName: 'ReactTestUtils',
+    isRenderer: false,
+    label: 'react-test-utils',
+    manglePropertiesOnProd: false,
+    name: 'react-test-utils',
```

Comment:
If you run build, you will see that there’s a single `react-test-utils.development.js` left in the `/build/` folder while everything else is neatly put into folders. This is probably because `name` corresponds to npm name, but there is no top-level `/packages/react-test-utils/` so it doesn’t understand where to take `package.json` from, and skips it. (I admit this behavior is confusing.)

The rule of thumb is that nothing should end up in top-level `/build/` folder, and that `/build/packages/` should be all publish-able “as is” after build. I don’t think this is currently publish-able as is because test utils don’t actually make their way into the `react-dom` package.

You probably want to put `'react-dom/test-utils'` in this field instead. This will ensure it gets copied into `/build/packages/react-dom/` instead. You can see we are using the same trick for `react-dom/server`.

However, I’m not sure how it will end up on npm. Don’t we need to add a `test-utils.js` file to `/packages/react-dom/` which points to the entry point, similarly to how we have `server.js` in `/packages/react-dom/`? And we’d probably want to add more entries to `"files"` in `/packages/react-dom/package.json` so that the new files actually end up in the package. Otherwise this PR does not really do anything to make it appear in the `react-dom` package.

I am slightly concerned about www. I would like to avoid deviating too much between how `ReactTestUtils` works in open source and in www. Currently, `ReactTestUtils` is a shim (`/scripts/rollup/shims/facebook-www/`) that re-exports a hidden value from `ReactDOM`. If we’re separating them in open source, can we also separate them in www?

The way I see it, we could do it by removing `ReactTestUtils` from the secret exports of `ReactDOMFBEntry.js` and `ReactDOMFiberFBEntry.js`, removing the `ReactTestUtils.js` shim from `/scripts/rollup/shims/facebook-www/`, and instead adding a `FB_DEV` target to this bundle. We would need to make sure it imports `require('react-dom')` as an external so that it doesn’t duplicate `ReactDOM` inside.

I have a similar concern about duplication in the open source build too (which this PR implements). If you build it and look inside the bundle, it completely duplicates `ReactDOM` inside. However this means that components using `findDOMNode` will import them from `react-dom` that is different from the one `react-dom/test-utils` is using. If you look at 15.5, [`react-dom/test-utils`](https://unpkg.com/react-dom@15.5.4/test-utils.js) intentionally shared the implementation with `react-dom` itself because otherwise it can’t function correctly. So I think the need to declare `react-dom` as an external affects not only FB bundle, but open source bundle as well.

Does this make sense?